crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -58,16 +58,27 @@ impl BlobStore for DiskFileBlobStore {
     }
 
     fn delete(&self, tx: B256) -> Result<(), BlobStoreError> {
-        self.inner.delete_one(tx)?;
+        let mut files_to_delete = self.inner.files_to_delete.write();
+        files_to_delete.insert(tx);
         Ok(())
     }
 
     fn delete_all(&self, txs: Vec<B256>) -> Result<(), BlobStoreError> {
-        if txs.is_empty() {
-            return Ok(())
-        }
+        self.inner.files_to_delete.write().extend(txs);
+        Ok(())
+    }
 
-        self.inner.delete_many(txs)?;
+    fn cleanup(&self) -> Result<(), BlobStoreError> {
+        let txs_to_delete = {
+            let mut files_to_delete = self.inner.files_to_delete.write();
+            std::mem::take(&mut *files_to_delete)
+        };
+        for tx in txs_to_delete {
+            let file_path = self.inner.blob_disk_file(tx);
```

Comment:
in the meanwhile, we should count the files to delete, and the total bytes to remove, for the first one, we can simply count it as `num+=1`, for the later one, maybe we can get the bytes with the file size?, eg:

```rust
if let Ok(metadata) = fs::metadata(&path) {     
    sub += metadata.len() as usize;             
}
```

Reply:
Done!